std/algorithm/searching.d

```diff
@@ -1788,6 +1818,41 @@ if (isRandomAccessRange!R1 && hasLength!R1 && hasSlicing!R1 && isBidirectionalRa
     {
         return haystack[haystack.length .. haystack.length];
     }
+    // Optimization in case the ranges are both SortedRanges.
+    // Binary search can be used to find the first occurence
+    // of the first element of the needle in haystack.
+    // When it is found O(walklength(needle)) steps are performed.
+    // 8829 enhancement
+    import std.range;
+    import std.algorithm.comparison: mismatch;
+    static if (is(R1 == R2)
+            && is(R1 : SortedRange!TT, TT)
+            && pred == "a == b")
+    {
+        auto needleFirstElem = needle[0];
+        auto partitions      = haystack.trisect(needleFirstElem);
+        auto firstElemLen    = partitions[1].length;
+        size_t count         = 0;
+
+        if (firstElemLen == 0)
+            return haystack[$ .. $];
+
+        while (needle.front() == needleFirstElem)
+        {
+            needle.popFront();
+            ++count;
+
+            if (count > firstElemLen)
+                return haystack[$ .. $];
+        }
+
+        auto m = mismatch(partitions[2], needle);
+
+        if (m[1].empty)
+            return haystack[partitions[0].length + partitions[1].length - count .. $];
+        if (!m[1].empty)
```

Comment:
This condition is the complement of the previous one, which if passes will terminate the flow. Remove.